extensions/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/DataPlaneFrameworkExtension.java

```diff
@@ -78,6 +89,12 @@ public void initialize(ServiceExtensionContext context) {
         transferServiceRegistry.registerTransferService(transferService);
         context.registerService(TransferServiceRegistry.class, transferServiceRegistry);
 
+        var numThreads = context.getSetting(TRANSFER_THREADS, DEFAULT_TRANSFER_THREADS);
+        var executorService = Executors.newFixedThreadPool(numThreads);
+        var executorContainer = new DataTransferExecutorServiceContainer(
+                executorInstrumentation.instrument(executorService, "Data plane transfers"));
```

Comment:
how do we ensure these valuable executor service instrumentations remains there and not replaced by non-instrumented executors ? Do you plan to add some tests to validate expected metrics are being produced or not ?

Reply:
Added to #161 so we keep track of this